code/modules/ruins/lavalandruin_code/tomb.dm

```diff
@@ -0,0 +1,181 @@
+//ENEMIES//
+
+/mob/living/simple_animal/hostile/cryptguard
+	name = "statue"
+	desc = "An incredibly lifelike stone carving depicting a brave warrior."
+	icon = 'icons/obj/statue.dmi'
+	icon_state = "human_male"
+	icon_living = "human_male"
+	icon_dead = "human_male"
+	speak_emote = list("says")
+	health = 100
+	maxHealth = 100
+	melee_damage_lower = 30
+	melee_damage_upper = 30
+	attacktext = "slashes"
+	deathmessage = "crumbles to dust."
+	AIStatus = AI_OFF
+	status_flags = 0
+	anchored = TRUE
+	a_intent = INTENT_HARM
+	var/woke = FALSE //cannot take damage if true
+	var/swiping = FALSE
+	var/powerful = FALSE //the boss can move while setting up attacks, the small ones cannot
+
+/mob/living/simple_animal/hostile/cryptguard/AttackingTarget()
+	if(swiping == TRUE)
+		if(client)
+			to_chat(src, "<span class='warning'>You're already attacking!</span>")
+		return
+	swiping = TRUE
+	var/turf/T = get_turf(target)
+	var/atktype = pick("cleave", "lunge")
+	switch(atktype)
+		if("lunge")
+			var/turf/Tstep = get_step(T, dir)
+			var/turf/Tstepstep = get_step(Tstep, dir)
+			new /obj/effect/temp_visual/attackwarn(T, src)
+			new /obj/effect/temp_visual/attackwarn(Tstep, src)
+			new /obj/effect/temp_visual/attackwarn(Tstepstep, src)
+		if("cleave")
+			var/turf/src_turf = get_turf(src)
+			var/dir_to_target = get_dir(src_turf, get_turf(target))
+			var/static/list/cleave_angles = list(0, -45, 45) //YOINK THANKS CLEAVING SAW YOU FUCK LOL
+			for(var/i in cleave_angles)
+				var/turf/Tcleave = get_step(src_turf, turn(dir_to_target, i))
+				new /obj/effect/temp_visual/attackwarn(Tcleave, src)
+	visible_message("<span class='warning'>[src] winds up to [atktype] with it's greatsword...</span>", "<span class='notice'>You begin to wind up a [atktype]...</span>")
+
+/obj/effect/temp_visual/attackwarn
+	name = "incoming attack"
+	desc = "a deer in headlights..."
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "attackarea"
+	layer = BELOW_MOB_LAYER
+	duration = 10
+	var/mob/living/simple_animal/hostile/cryptguard/createdby
+
+/obj/effect/temp_visual/attackwarn/Initialize(mapload, createdby)
+    ..(mapload)
+    src.createdby = createdby
+
+/obj/effect/temp_visual/attackwarn/Destroy()
+	var/turf/T = get_turf(src)
+	for(var/mob/living/L in T)
+		to_chat(L, "<span class='danger'>You are hit by [createdby]'s greatsword!</span>")
+		L.adjustBruteLoss(30)
+	createdby.swiping = FALSE
+	..()
+//hit
+
+/mob/living/simple_animal/hostile/cryptguard/Move()
+	if(swiping == TRUE && powerful == FALSE)
+		if(client)
+			to_chat(src, "<span class='warning'>You can't move while swinging a sword like this!</span>")
+		return
+	..()
+
+
+/mob/living/simple_animal/hostile/cryptguard/proc/awaken()
+	if(!istype(src, /mob/living/simple_animal/hostile/cryptguard))
```

Comment:
can you explain this check

Reply:
yeah, only crypt guards have the woke proc so we really really wanna make sure it is a crypt guard

u-unless the proc being for cryptguards only is that...

Reply:
If only they have a proc when would this check ever fail? It would runtime at mob trying to call awaken because the mob in question doesn't have that proc.

If anything, the check would be above this (as in when it gets called).

Reply:
ok. thanks. i'll remove the check!